src/main/helper/ApiFormatter.ts

```diff
@@ -104,11 +105,23 @@ export function toApiFormat(caseItem: CaseWithId): UpdateCaseBody {
   };
 }
 
-function formatDate(dobDate: CaseDate) {
-  return dobDate ? `${dobDate.year}-${dobDate.month}-${dobDate.day}` : null;
-}
+export const formatDate = (dobDate: CaseDate): string => {
+  if (isDateEmpty(dobDate)) {
+    return null;
+  }
 
-function parseDateFromString(dobDate: string): CaseDate {
+  return (
+    dobDate.year +
+    '-' +
+    (dobDate.month.match(/^\d$/) ? '0' : '') +
+    dobDate.month +
+    '-' +
+    (dobDate.day.match(/^\d$/) ? '0' : '') +
+    dobDate.day
+  );
+};
```

Comment:
Template literal would read nicer here, also if the intention of the match is to add a preceding 0. you can use `padStart(2, '0')` to make sure its length is always 2. (ie, `dobDate.month.padStart(2, '0')`)

```return `${dobDate.year}-${dobDate.month.padStart(2, '0')}-${dobDate.day.padStart(2, '0')}` ```